lib/Doctrine/ConfigurationFactory.php

```diff
@@ -55,4 +58,17 @@ private function configureProxies(Configuration $configuration) {
         : AbstractProxyFactory::AUTOGENERATE_NEVER
     );
   }
+
+  private function configureCache(Configuration $configuration) {
+    if ($this->is_dev_mode) {
+      $cache = new ArrayCache();
+    } else {
+      $cache_dir = Env::$cache_path . '/doctrine';
+      $cache = new FilesystemCache($cache_dir);
```

Comment:
I'm bit worried about using `FilesystemCache`. I'm not aware of any mechanism for cleaning $cache_dir on update so I'm a bit afraid that we use result cache somewhere and we change something in some entity and cached result could be incorrect. Can that happen? Maybe we could add plugin version into namespace or just use ArrayCache also for production.

Reply:
Hmm, one could also think query cache uses hash of the query as a key and result cache... I'm not sure, I have to check. It's possible this cache can live even between releases but maybe it's better to invalidate it. If we do it, we may have a problem of never cleaning it 🤔

As for cleaning I guess we may have that problem with Twig at the moment - never cleaning anything. That's not good because if I add the version there the cache will only grow. I'll check it but staying with `ArrayCache` works as well I guess.

Reply:
I think query cache should be fine. It caches [only DQL -> SQL transformation](https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/caching.html#query-cache). The [result cache](https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/caching.html#result-cache) could be a tricky one here. Developer have to manually select which query to cache and take care about invalidation. So maybe we could keep some list of cached ids and do invalidation on update. I think that we don't use any type of cache at the moment so staying with `ArrayCache` seems reasonable for now.
Good point about Twig. Maybe we could generate this cache build time in the future. I'm not sure if is is possible though.

Reply:
Sounds good. Fixed using `ArrayCache` at the moment.